teamworkApp/lib/evaluateAnswers.py

```diff
@@ -3,47 +3,39 @@
 
 DB = 'db/development.sqlite3'
 
+QUESTIONS = [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17]
+
 Style = IntEnum('Style', 'Contributor, Collaborator, Communicator, Challenger', start=0)
 Answer_Value = IntEnum(
-    'Answer_Value', 
+    'Answer_Value',
     '1234 1243 1324 1342 1423 1432 2134 2143 2314 2341 2413 2431 3124 3142 3214 3241 3412 3421 4123 4132 4213 4231 4312 4321',
     start=0
 )
 
-# the order of answers repeat every 4 questions so just the index is (question_num % 4) 
+# the order of answers repeat every 4 questions so just the index is (question_num % 4)
 Questions = {
     0: (Style.Contributor, Style.Collaborator, Style.Communicator, Style.Challenger),
     1: (Style.Collaborator, Style.Communicator, Style.Challenger, Style.Contributor),
     2: (Style.Communicator, Style.Challenger, Style.Contributor, Style.Collaborator),
     3: (Style.Challenger, Style.Contributor, Style.Collaborator, Style.Communicator),
 }
 
-def get_students_answers(student_id):
-    """return list of complete answers for a given student"""
-    conn = sqlite3.connect(DB)
-    c = conn.cursor()
-
-    c.execute(
-        'SELECT * FROM answers WHERE student_id=?', 
-        [student_id,],
-    )
-    rows = c.fetchall()
-
-    conn.commit()
-    conn.close()
 
-    return rows
-
-def find_scores(student_id):
-    """Returns a student's scores for each of the possible styles in a tuple"""
-    answers = get_students_answers(student_id)
-    questions = [answer[5] for answer in answers]
-    values = [Answer_Value(answer[1]).name for answer in answers]
+def find_scores(expanded_responses):
+    """Given a list of expanded responses (i.e. '1234', '3214'...), find_scores
+        will follow the scoring pattern outlined in Glenn Parker's
+        teamwork survey.
+        output: a tuple of four scores reflecting the four styles of the
+            teamwork survey, ie
+            (Contributor, Collaborator, Communicator, Challenger)"""
     scores = [0] * len(Style)
-    
-    for question in range(len(questions)):
+    # loop through all the numbers corresponding to questions
+    for question in range(len(QUESTIONS)):
+        #find the order in which to score the expanded response
         order = Questions[question % 4]
         for i, style in enumerate(order):
-            scores[style.value] += int(values[question][i])
-
-    return tuple(scores)
\ No newline at end of file
+            # find the style and add it to the ongoing incrementation
+            # of the score
+            scores[style.value] += int(expanded_responses[question][i])
+    #(contributor, collaborator, communicator, challenger)
+    return tuple(scores)
```

Comment:
How does this handle a non-complete dataset? Do we want to throw an error, accept it, or some other option?

Reply:
Could you elaborate on what you mean by incomplete dataset?

I agree that we should be checking for incomplete things, but I'm not quite sure what an incomplete dataset will look like with this wording?

Reply:
I was thinking about what would happen if not all of the questions were answered.

Reply:
Hmm I suppose that's a matter of enforcing where the .csv came from because in the survey I made, every answer is required. So should we somehow insist that data only come from that google form?

Reply:
I think requiring data to come from the survey is acceptable at this point, especially considering we're bumping up against the end of the project. We'll just want to notate it somewhere in the ReadMe